### ISSUE TITLE
Impose colore on tag creation

### ISSUE THREAD

**Issue by barsan-md** [yairEO/tagify#666]
I'm trying to create something similar to [Easy to customize](https://yaireo.github.io/tagify/#section-different-look) (anchor does not work in firefox on this specific section) example with a key difference: I have two or more (+) buttons, each adds the tag of a different color.

```HTML
<input type="text" id="menu-tags-input" class="menu-tags" />
<button id="add-whitelist-tag" class="btn btn-success">+</button>
<button id="add-blacklist-tag" class="btn btn-danger">+</button>
```
I thought I could to create some information in the empty tag that is created by the button and read it in the actual tag after it is populated with valid data. I tried using style or dataset attributed but both are erased when the data is selected. I also tried to listen to the events 'edit:beforeUpdate' and 'edit:updated' to see if I have some before after reference (I have `editTags: false` though) but they do not fire in this case.
```JavaScript
let input = document.getElementById('menu-tags-input')
    tagify = new Tagify(input, {...}),
    whitelistTags = document.getElementById('add-whitelist-tag')

whitelistTags.addEventListener('click', function () {
    let tagData = { value: '' },
      tagElm = tagify.createTagElem(tagData)
    tagElm.style = '...' // attempt 1
    tagElm.dataset.list = 'whitelist' // attempt 2
    tagify.appendTag(tagElm)
    tagify.editTag(tagElm, { skipValidation: true })
  })
```

What would be the best way to achieve this?

**Comment by yairEO** [comment]
Why don'y you split it into 2 separate tag components? Your back-end would have to split it somehow, using some *identifier* to know which *tag* is for *whitelist* and which for *blacklist*..

You can use the `transformTag` (function) *setting* to add mutate the tag's data, just before creation, and in that function you can check which button was clicked (whitelist/blacklist) and set some identifier (for a color) which you could use in the CSS or directly as a `style` property:

[I've made you a demo](https://jsbin.com/xevijop/edit?html,css,js,output)

```js
var tagify = new Tagify(inputElement, {
	transformTag(tagData){
		// here you must know which button created this tag
		tagData.type = ...
	}
}

// 
```

Or use `addEmptyTag` method to add custom identifiers to your tags and then target that in the CSS:

```js
whitelistTags.addEventListener('click', function () {
	tagify.addEmptyTag({ type:'whitelist' })
}
```

---
[Related question](https://github.com/yairEO/tagify/issues/644)

**Comment by barsan-md** [comment]
Thanks a lot! I didn't want to split because I wanted the tags to be either in one list or another, not both.

**Comment by barsan-md** [comment]
Sorry for reopening this issue but I found a bug.
https://jsbin.com/difagowada/edit?html,css,js,output
I just added a whitelist to the example you provided.
`whitelist: ['aa', 'bb']`

Now if you try to a tag using the "salmon" plus, digit aa or bb and do **not** select the element from the dropdown, just press enter. It will be blue because it lost the attribute `color` and blue is the default.

Thanks
